pip_audit/_format/markdown.py

```diff
@@ -0,0 +1,135 @@
+            output += skipped_deps_output
+        return output
+
+    def _format_vuln_results(
+        self,
+        result: Dict[service.Dependency, List[service.VulnerabilityResult]],
+        fixes: List[fix.FixVersion],
+    ) -> str:
+        header = "Name | Version | ID | Fix Versions"
+        border = "--- | --- | --- | ---"
+        if fixes:
+            header += " | Applied Fix"
+            border += " | ---"
+        if self.output_desc:
+            header += " | Description"
+            border += " | ---"
+
+        vuln_rows: List[str] = []
+        for dep, vulns in result.items():
+            if dep.is_skipped():
+                continue
+            dep = cast(service.ResolvedDependency, dep)
+            applied_fix = next((f for f in fixes if f.dep == dep), None)
+            for vuln in vulns:
+                vuln_rows.append(self._format_vuln(dep, vuln, applied_fix))
+
+        if not vuln_rows:
+            return str()
+
+        return header + os.linesep + border + os.linesep + os.linesep.join(vuln_rows)
+
+    def _format_vuln(
+        self,
+        dep: service.ResolvedDependency,
+        vuln: service.VulnerabilityResult,
+        applied_fix: Optional[fix.FixVersion],
+    ) -> str:
+        vuln_text = (
+            f"{dep.canonical_name} | {dep.version} | {vuln.id} | "
+            f"{self._format_fix_versions(vuln.fix_versions)}"
+        )
+        if applied_fix is not None:
+            vuln_text += f" | {self._format_applied_fix(applied_fix)}"
+        if self.output_desc:
+            vuln_text += f" | {vuln.description}"
+        return vuln_text
+
+    def _format_fix_versions(self, fix_versions: List[Version]) -> str:
+        return ",".join([str(version) for version in fix_versions])
+
+    def _format_applied_fix(self, applied_fix: fix.FixVersion) -> str:
+        if applied_fix.is_skipped():
+            applied_fix = cast(fix.SkippedFixVersion, applied_fix)
+            return (
+                f"Failed to fix {applied_fix.dep.canonical_name} ({applied_fix.dep.version}): "
+                f"{applied_fix.skip_reason}"
+            )
+        applied_fix = cast(fix.ResolvedFixVersion, applied_fix)
+        return (
+            f"Successfully upgraded {applied_fix.dep.canonical_name} ({applied_fix.dep.version} "
+            f"=> {applied_fix.version})"
+        )
+
+    def _format_skipped_deps(
+        self, result: Dict[service.Dependency, List[service.VulnerabilityResult]]
+    ) -> str:
+        header = "Name | Skip Reason"
+        border = "--- | ---"
+
+        skipped_dep_rows: List[str] = []
+        for dep, _ in result.items():
+            if dep.is_skipped():
+                dep = cast(service.SkippedDependency, dep)
+                skipped_dep_rows.append(self._format_skipped_dep(dep))
+
+        if not skipped_dep_rows:
+            return str()
+
+        return header + os.linesep + border + os.linesep + os.linesep.join(skipped_dep_rows)
```

Comment:
GFM/CommonMark specify a "line ending" as either `\n` or `\r\n`, so I think we can build this as a heredoc instead. I'll make that change.